tests/test_procrustes_step.py

```diff
@@ -0,0 +1,162 @@
+        self.device = torch.device("cuda") if torch.cuda.is_available() else torch.device("cpu")
+
+    def _procrustes_objective(self, Q: torch.Tensor) -> torch.Tensor:
+        """Helper function to compute Procrustes objective ||Q^H Q - I||_F^2."""
+        return torch.linalg.matrix_norm(Q.H @ Q - torch.eye(Q.size(0), dtype=Q.dtype, device=Q.device), ord="fro") ** 2
+
+    def test_improves_orthogonality_simple_case(self) -> None:
+        """Test that procrustes_step doesn't worsen orthogonality for a simple case."""
+
+        # Make a SPD non-orthogonal matrix
+        Q = torch.randn(2, 2, device=self.device, dtype=torch.float32)
+        Q = Q @ Q.T
+
+        initial_obj = self._procrustes_objective(Q)
+
+        procrustes_step(Q, max_step_size=1 / 16)
+
+        final_obj = self._procrustes_objective(Q)
+
+        self.assertLessEqual(final_obj.item(), initial_obj.item() + 1e-6)
+
+    def test_modifies_matrix_in_place(self) -> None:
+        """Test that procrustes_step modifies the matrix in place."""
+        Q = torch.randn(3, 3, device=self.device)
+        Q_original_id = id(Q)
+
+        procrustes_step(Q, max_step_size=1 / 16)
+
+        self.assertEqual(id(Q), Q_original_id)
+
+    @parameterized.parameters(
+        (8,),
+        (128,),
+        (1024,),
+    )
+    def test_minimal_change_when_already_orthogonal(self, size: int) -> None:
+        """Test that procrustes_step makes minimal changes to an already orthogonal matrix."""
+        # Create an orthogonal matrix using QR decomposition
+        A = torch.randn(size, size, device=self.device, dtype=torch.float32)
+        with fp32_matmul_precision("highest"):
+            Q, _ = torch.linalg.qr(A)
+
+        initial_obj = self._procrustes_objective(Q)
+
+        procrustes_step(Q, max_step_size=1 / 16)
+
+        final_obj = self._procrustes_objective(Q)
+
+        # For already orthogonal matrices, the objective should remain small
+        self.assertLess(final_obj.item(), 1e-5)
+        self.assertLess(final_obj.item(), initial_obj.item() + 1e-5)
+
+    def test_handles_small_norm_gracefully(self) -> None:
+        """Test that procrustes_step handles matrices with small R norm improvement."""
+        # Create a matrix very close to orthogonal
+        A = torch.randn(3, 3, device=self.device, dtype=torch.float32)
+        with fp32_matmul_precision("highest"):
+            Q, _ = torch.linalg.qr(A)
+        # Add tiny perturbation
+        Q += 1e-10 * torch.randn_like(Q, dtype=torch.float32)
+
+        initial_obj = self._procrustes_objective(Q)
+
+        procrustes_step(Q, max_step_size=1 / 16)
+
+        final_obj = self._procrustes_objective(Q)
+
+        self.assertLess(final_obj.item(), 1e-6)
+        self.assertLess(final_obj.item(), initial_obj.item() + 1e-6)
+
+    @parameterized.parameters(
+        (1 / 64,),
+        (1 / 32,),
+        (1 / 16,),
+        (1 / 8,),
+    )
+    def test_different_step_sizes(self, max_step_size: float) -> None:
```

Comment:
Suggestion: include designated behavior in test name. e.g. test_xxx_what_smaller_than_what.
Same for all other test names.